src/Neo/Network/P2P/RemoteNode.cs

```diff
@@ -76,13 +76,14 @@ internal class Relay { public IInventory Inventory; }
         /// <param name="connection">The underlying connection object.</param>
         /// <param name="remote">The address of the remote node.</param>
         /// <param name="local">The address of the local node.</param>
-        public RemoteNode(NeoSystem system, LocalNode localNode, object connection, IPEndPoint remote, IPEndPoint local)
+        /// <param name="config">P2P settings.</param>
+        public RemoteNode(NeoSystem system, LocalNode localNode, object connection, IPEndPoint remote, IPEndPoint local, ChannelsConfig config)
             : base(connection, remote, local)
         {
             this.system = system;
             this.localNode = localNode;
-            _knownHashes = new HashSetCache<UInt256>(system.MemPool.Capacity * 2 / 5);
-            _sentHashes = new HashSetCache<UInt256>(system.MemPool.Capacity * 2 / 5);
+            _knownHashes = new HashSetCache<UInt256>(Math.Max(1, config.MaxKnownHashes / 10), 10);
+            _sentHashes = new HashSetCache<UInt256>(Math.Max(1, config.MaxKnownHashes / 10), 10);
```

Comment:
@shargon, in such cases, I think it would be better to set some logic that limits the minimum value of MaxKnownHashes to `10` right on the config.
Instead of using the Math.Max.

But both are good to me. But I do not like much this here.